subsys/bluetooth/audio/mcc.c

```diff
@@ -1538,48 +1576,83 @@ static uint8_t discover_mcs_char_func(struct bt_conn *conn,
 			BT_DBG("Parent Group Object, UUID: %s", bt_uuid_str(chrc->uuid));
 			cur_mcs_inst->parent_group_obj_id_handle = chrc->value_handle;
 			cur_mcs_inst->parent_group_obj_sub_params.disc_params = NULL;
+			if (chrc->properties & BT_GATT_CHRC_NOTIFY) {
+				cur_mcs_inst->parent_group_obj_sub_params.value =
+					BT_GATT_CCC_NOTIFY;
+			}
 		} else if (!bt_uuid_cmp(chrc->uuid, BT_UUID_MCS_CURRENT_GROUP_OBJ_ID)) {
 			BT_DBG("Group Object, UUID: %s", bt_uuid_str(chrc->uuid));
 			cur_mcs_inst->current_group_obj_id_handle = chrc->value_handle;
 			cur_mcs_inst->current_group_obj_sub_params.disc_params = NULL;
+			if (chrc->properties & BT_GATT_CHRC_NOTIFY) {
+				cur_mcs_inst->current_group_obj_sub_params.value =
+					BT_GATT_CCC_NOTIFY;
+			}
 #endif /* CONFIG_BT_MCC_OTS */
 		} else if (!bt_uuid_cmp(chrc->uuid, BT_UUID_MCS_PLAYING_ORDER)) {
 			BT_DBG("Playing Order, UUID: %s", bt_uuid_str(chrc->uuid));
 			cur_mcs_inst->playing_order_handle = chrc->value_handle;
 			cur_mcs_inst->playing_order_sub_params.disc_params = NULL;
+			if (chrc->properties & BT_GATT_CHRC_NOTIFY) {
+				cur_mcs_inst->playing_order_sub_params.value = BT_GATT_CCC_NOTIFY;
+			}
 		} else if (!bt_uuid_cmp(chrc->uuid, BT_UUID_MCS_PLAYING_ORDERS)) {
 			BT_DBG("Playing Orders supported, UUID: %s", bt_uuid_str(chrc->uuid));
 			cur_mcs_inst->playing_orders_supported_handle = chrc->value_handle;
 		} else if (!bt_uuid_cmp(chrc->uuid, BT_UUID_MCS_MEDIA_STATE)) {
 			BT_DBG("Media State, UUID: %s", bt_uuid_str(chrc->uuid));
 			cur_mcs_inst->media_state_handle = chrc->value_handle;
 			cur_mcs_inst->media_state_sub_params.disc_params = NULL;
+			if (chrc->properties & BT_GATT_CHRC_NOTIFY) {
+				cur_mcs_inst->media_state_sub_params.value = BT_GATT_CCC_NOTIFY;
+			}
 		} else if (!bt_uuid_cmp(chrc->uuid, BT_UUID_MCS_MEDIA_CONTROL_POINT)) {
 			BT_DBG("Media Control Point, UUID: %s", bt_uuid_str(chrc->uuid));
 			cur_mcs_inst->cp_handle = chrc->value_handle;
 			cur_mcs_inst->cp_sub_params.disc_params = NULL;
+			if (chrc->properties & BT_GATT_CHRC_NOTIFY) {
+				cur_mcs_inst->cp_sub_params.value = BT_GATT_CCC_NOTIFY;
+			}
 		} else if (!bt_uuid_cmp(chrc->uuid, BT_UUID_MCS_MEDIA_CONTROL_OPCODES)) {
 			BT_DBG("Media control opcodes supported, UUID: %s",
 			       bt_uuid_str(chrc->uuid));
 			cur_mcs_inst->opcodes_supported_handle = chrc->value_handle;
 			cur_mcs_inst->opcodes_supported_sub_params.disc_params = NULL;
+			if (chrc->properties & BT_GATT_CHRC_NOTIFY) {
+				cur_mcs_inst->opcodes_supported_sub_params.value =
+					BT_GATT_CCC_NOTIFY;
+			}
 #ifdef CONFIG_BT_MCC_OTS
 		} else if (!bt_uuid_cmp(chrc->uuid, BT_UUID_MCS_SEARCH_CONTROL_POINT)) {
 			BT_DBG("Search control point, UUID: %s", bt_uuid_str(chrc->uuid));
 			cur_mcs_inst->scp_handle = chrc->value_handle;
 			cur_mcs_inst->scp_sub_params.disc_params = NULL;
+			if (chrc->properties & BT_GATT_CHRC_NOTIFY) {
+				cur_mcs_inst->scp_sub_params.value = BT_GATT_CCC_NOTIFY;
+			}
```

Comment:
Since `do_subscribe` actually does `sub_params->value = BT_GATT_CCC_NOTIFY;` why may want to remove that, so that we don't overwrite this value. It'll work the same either way, but there's no need to overwrite the value in `do_subscribe` anymore.